x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/indicator_match/threat_mapping/get_allowed_fields_for_terms_query.ts

```diff
@@ -14,30 +14,25 @@ const allowedFieldTypes = ['keyword', 'constant_keyword', 'wildcard', 'ip'];
  * Return map of fields allowed for term query
  */
 export const getAllowedFieldForTermQueryFromMapping = (
-  indexMapping: IndicesGetFieldMappingResponse
+  fieldsCapsResponse: FieldCapsResponse,
+  fields: string[]
 ): Record<string, boolean> => {
-  const result: Record<string, boolean> = {};
-  const notAllowedFields: string[] = [];
+  const fieldsCaps = fieldsCapsResponse.fields;
 
-  const indices = Object.values(indexMapping);
-  indices.forEach((index) => {
-    Object.entries(index.mappings).forEach(([field, fieldValue]) => {
-      Object.values(fieldValue.mapping).forEach((mapping) => {
-        const fieldType = mapping?.type;
-        if (!fieldType) return;
+  const availableFields = fields.filter((field) => {
+    const fieldCaps = fieldsCaps[field];
 
-        if (allowedFieldTypes.includes(fieldType) && !notAllowedFields.includes(field)) {
-          result[field] = true;
-        } else {
-          notAllowedFields.push(field);
-          // if we the field allowed in one index, but not allowed in another, we should delete it from result
-          delete result[field];
-        }
-      });
+    const isAllVariationsAllowed = Object.values(fieldCaps).every((fieldCapByType) => {
+      return allowedFieldTypes.includes(fieldCapByType.type);
```

Comment:
nit: converting `allowedFieldTypes` to set, will avoid traversing this array in each iteration

```suggestion
      return allowedFieldTypesSet.has(fieldCapByType.type);
```